### PR TITLE
feat(fastly_integration): implement resource and documentation

### PR DESCRIPTION
**Pull request by darin-nee**
This PR implements a new `fastly_integration` resource.

### CODE REVIEW

File: docs/resources/integration.md
Comment (Integralist) [inline docs/resources/integration.md]: Out of interest, what is the workflow like for a user who sets up this new resource but _doesn't_ then confirm via email? Will the Fastly API eventually return an API error that will result in a new plan diff when the user runs `terraform plan`?

File: docs/resources/integration.md
Comment (Integralist) [inline docs/resources/integration.md:18]: I ❤️ the detailed examples

File: fastly/resource_fastly_integration.go
Comment (Integralist) [inline fastly/resource_fastly_integration.go:44]: The following suggested edit adds validation for the allowed inputs (it will also require the following import):

```
"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
```

```suggestion
				Description: "Type of the integration. One of: `mailinglist`, `microsoftteams`, `newrelic`, `pagerduty`, `slack`, `webhook`.",
				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(
					[]string{"mailinglist", "microsoftteams", "newrelic", "pagerduty", "slack", "webhook"},
					false,
				)),
```

File: fastly/resource_fastly_integration.go
Comment (Integralist) [inline fastly/resource_fastly_integration.go]: This might seem a bit silly but I'm very used to users not reading the docs and just clicking through to the schema and missing any examples we've written, so in this case, as `config` is tightly coupled to `type` (i.e. if a user reads the description alone, will they know what value to provide) I think this additional comment might be useful...

```suggestion
				Description: "Configuration specific to the integration `type` (see documentation examples).",
```

**Review by Integralist — APPROVED** [review]
This is great!

I'm approving to unblock, but I do have a couple of suggested edits...

File: docs/resources/integration.md
Comment (darin-nee) [reply docs/resources/integration.md]: Good question. It would likely be confusing for users who don't know about the extra step. To improve that experience, I added a diagnostic warning that will appear after applying or refreshing when there's an unconfirmed mailing list integration.

File: docs/resources/integration.md
Comment (darin-nee) [reply docs/resources/integration.md:18]: Thanks!

File: docs/resources/integration.md
Comment (darin-nee) [reply docs/resources/integration.md]: The Fastly API does not eventually return an API error; that resource just exists with one of several unconfirmed statuses `needs_confirmation`, `confirmation_pending`, or `confirmation_expired`. A further potential enhancement would be for Fastly Terraform provider to request a confirmation email automatically and save the user from having to visit a UI, which would simplify the workflow if the Terraform user doesn't have a Fastly login. There's a little complexity with where to put such a hook though to deduplicate the requests per email address for the case when there are multiple mailing list integrations with the same address.

Comment (Integralist) [inline docs/resources/integration.md]: I'm now thinking we should move this to a note in the documentation itself rather than the code example (see my suggested edit for `templates/resources/integration.md.tmpl`)
```suggestion
```

File: templates/resources/integration.md.tmpl
Comment (Integralist) [inline templates/resources/integration.md.tmpl:11]:
```suggestion
Provides a Fastly integration. Fastly will use these integrations to send you notifications and alerts related to the Fastly product. Mailing list, Microsoft Teams, New Relic, PagerDuty, Slack, and webhook are supported as integrations.

~> **IMPORTANT:** The mailing list integrations require confirmation. To send a confirmation email and verify integration status, after applying changes using Terraform, please visit https://manage.fastly.com/observability/alerts/integrations
```

This will be rendered in a nice yellow box with appropriate emoji icon (example: https://registry.terraform.io/providers/fastly/fastly/latest/docs/resources/secretstore)

**Review by Integralist — CHANGES_REQUESTED** [review]
LGTM. Just one requested change please before we merge...

File: examples/resources/integration_basic_usage.tf
Comment (Integralist) [inline examples/resources/integration_basic_usage.tf]: Sorry, made suggested edit to the wrong file

```suggestion
```

File: docs/resources/integration.md
Comment (darin-nee) [inline docs/resources/integration.md:12]:
```suggestion

~> **IMPORTANT:** The mailing list integrations require confirmation. To send a confirmation email and verify integration status, after applying changes using Terraform, please visit https://manage.fastly.com/observability/alerts/integrations

```

**Review by Integralist — APPROVED** [review]

**Pull request merged**
